src/main/java/uk/gov/hmcts/reform/migration/service/ChildrenDataMigrationService.java

```diff
@@ -0,0 +1,148 @@
+package uk.gov.hmcts.reform.migration.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.domain.common.Address;
+import uk.gov.hmcts.reform.domain.common.CollectionEntry;
+import uk.gov.hmcts.reform.domain.common.TelephoneNumber;
+import uk.gov.hmcts.reform.fpl.domain.CaseData;
+import uk.gov.hmcts.reform.fpl.domain.OldChildren;
+import uk.gov.hmcts.reform.fpl.domain.OldChild;
+import uk.gov.hmcts.reform.fpl.domain.Child;
+import uk.gov.hmcts.reform.fpl.domain.common.ChildParty;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.StringUtils.defaultIfBlank;
+import static org.springframework.util.ObjectUtils.isEmpty;
+
+@Slf4j
+@Component
+public class ChildrenDataMigrationService implements DataMigrationService<CaseData> {
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Override
+    public Predicate<CaseDetails> accepts() {
+        return caseDetails -> caseDetails != null && caseDetails.getData() != null &&
+            !isEmpty(caseDetails.getData().get("children"));
+    }
+
+    @Override
+    public CaseData migrate(Map<String, Object> data) {
+        CaseData caseData = objectMapper.convertValue(data, CaseData.class);
+
+        CaseData migratedCaseData = CaseData.builder()
+            .children1(migrateChildren(caseData.getChildren()))
+            .build();
+
+        log.info("new case details: {}", migratedCaseData);
```

Comment:
None of this code is being run by us in production. It will be packaged into a jar and run by devOps. They have security clearance to see logs and case info 👍